InfiniteLoopBookProject/src/com/st/lms/menu/Menu.java

```diff
@@ -0,0 +1,420 @@
+				}
+				break;
+			case 5:
+				List<String> updates = new ArrayList<String>();
+				updates.add("1) Change publisher's name.");
+				updates.add("2) Change publisher's address.");
+				updates.add("3) Change publisher's phone number.");
+				updates.add("9) Go back to Publisher Menu.");
+				System.out.println("Please enter the id number of the publisher you would like to update.");
+				publisherID = getValidID();
+				publisher = Service.getPublisher(publisherID);
+				if(publisher==null)
+					System.out.println("That publisher does not exist in our records!");
+				else {
+					System.out.println("What would you like to change?");
+					for(String update : updates)
+						System.out.println(update);
+					int temp = getValidOption(updates.size());
+					switch(temp) {
+						case 1:
+							System.out.println("Please enter the publisher's new name.");
+							publisherName = scan.nextLine();
+							publisher = new Publisher(publisherName, publisherID, publisher.getAddr(), publisher.getPhone());
+							Service.updatePublisher(publisherID, publisher);
+							break;
+						case 2:
+							System.out.println("Please enter the publisher's new address.");
+							publisherAddr = scan.nextLine();
+							publisher = new Publisher(publisher.getName(), publisherID, publisherAddr, publisher.getPhone());
+							Service.updatePublisher(publisherID, publisher);
+							break;
+						case 3:
+							System.out.println("Please enter the publisher's new phone number.");
+							publisherPhone = Long.parseLong(getValidPhone());
+							publisher = new Publisher(publisher.getName(), publisherID, publisher.getAddr(), publisherPhone);
+							Service.updatePublisher(publisherID, publisher);
+							break;
+						default:
+							break;
+					}
+				}
+			case EXIT:
+				System.out.println("Returning to main menu...");
+				break;
+			default:
+				break;
+			}
+		}while(input != EXIT);
+		run();
+	}
+	
+	private int getValidOption(int options) {//prompts user for a number less than options-1 or 9 for exit
+		System.out.println("Which option would you like to choose?");
+		String input;
+		int option;
+		do {
+			input = scan.nextLine();
+			try {
+				option = Integer.parseInt(input);
+				if((option!=9) && (option<1 || option>(options-1))) {
+					option = 0;
+					System.out.println("ERROR: That is not one of the options listed!");
+				}
+			}catch(NumberFormatException e) {
+				System.out.println("ERROR: Please enter a number from the options available!");
+				option = 0;
+			}
+		}while(option == 0);
+		return option;
+	}
+	
+	private int getValidID() { //prompts user until it receives a positive id number
+		String input;
+		int id;
+		
+		do {
+			input = scan.nextLine();
+			try {
+				id = Integer.parseInt(input);
+			}catch(NumberFormatException e){
+				id = 0;
+				System.out.println("ERROR: invalid id number. Please enter a positive integer.");
+			}
+		}while(!(id > 0));
+		return id;
+	}
+	
+	private String getValidPhone() {

```

Comment:
Like the previous PR I do like the fact that you check to see the user entered numbers and that it is 10 characters long.